src/PhpSpec/Process/ReRunner/WindowsPassthruReRunner.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Process\ReRunner;
+
+use PhpSpec\Process\Context\ExecutionContextInterface;
+use Symfony\Component\Process\PhpExecutableFinder;
+
+class WindowsPassthruReRunner extends PhpExecutableReRunner
+{
+    /**
+     * @var ExecutionContextInterface
+     */
+    private $executionContext;
+
+    /**
+     * @param PhpExecutableFinder $phpExecutableFinder
+     * @param ExecutionContextInterface $executionContext
+     * @return static
+     */
+    public static function withExecutionContext(PhpExecutableFinder $phpExecutableFinder, ExecutionContextInterface $executionContext)
+    {
+        $reRunner = new static($phpExecutableFinder);
+        $reRunner->executionContext = $executionContext;
+
+        return $reRunner;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isSupported()
+    {
+        return (php_sapi_name() == 'cli')
+            && $this->getExecutablePath()
+            && function_exists('passthru')
+            && (stripos(PHP_OS, "win") === 0);
+    }
+
+    public function reRunSuite()
+    {
+        $args = $_SERVER['argv'];
+        $command = $this->buildArgString() . '"' . $this->getExecutablePath().'" '.join(' ', array_map('escapeshellarg', $args));
```

Comment:
@ciaranmcnulty maybe he does not have spaces in his PHP executable path, meaning he is not affected by the issue I reported.
If removing the escaping solved an issue for him, it means there might be *another* issue which disappears when removing escaping. But this needs to be investigated then. Saying ``all the data they escape comes from internal processes which makes me a bit more comfortable about it`` is wrong:

- the PHP path does not come from internal stuff (it depends on the user setup)
- the escaping is here because of the needs of the shell interpreter, not just because of accepting untrusted input